### PR TITLE
Use fileCDN with video attachments

### PR DESCRIPTION
**Pull request by bpollman**
### 🔗 Issue Link
_Jira or Github issue link, if applicable._

### 🎯 Goal

We need a way for all attachment URL's to be intercepted and adjusted/signed. 

### 🛠 Implementation

This update adds FileCDN handling to all video attachment touch points primarily via the:
- `VideoPreviewLoader` for thumbnails
- Gallery -> `StreamVideoPlayer` and `VideoPlayerView`

Also make a few style & comment clean ups from previous FileCDN PR. 

### 🧪 Testing

Added basic tests

### 🎨 Changes

_Add relevant screenshots or videos showcasing the changes._

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

**Review by martinmitrevski — COMMENTED** [review]
Thanks again for your contribution! Looks good, but we just need to adjust the `VideoPreviewLoader`, so it doesn't have breaking changes (check the comment).

File: Sources/StreamChatSwiftUI/Utils/Common/VideoPreviewLoader.swift [hunk 1]
```diff
@@ -34,31 +34,43 @@ public final class DefaultVideoPreviewLoader: VideoPreviewLoader {
         NotificationCenter.default.removeObserver(self)
     }
 
-    public func loadPreviewForVideo(at url: URL, completion: @escaping (Result<UIImage, Error>) -> Void) {
+    public func loadPreviewForVideo(at url: URL, fileCDN: FileCDN, completion: @escaping (Result<UIImage, Error>) -> Void) {
```
[the remark this answers is not in the document]
Comment (bpollman) [reply Sources/StreamChatSwiftUI/Utils/Common/VideoPreviewLoader.swift:37]: No problem, will get it updated. Thanks for the prompt reviews!

[the remark this answers is not in the document]
Comment (bpollman) [reply Sources/StreamChatSwiftUI/Utils/Common/VideoPreviewLoader.swift]: Done! ✅

**Review by martinmitrevski — APPROVED** [review]
Looks good to me ✅

**Pull request merged**